src/Text/Markdown/SlamDown.purs

```diff
@@ -197,14 +199,53 @@ everythingM b i (SlamDown bs) = mconcat <$> traverse b' bs
 everything :: forall r. (Monoid r) => (Block -> r) -> (Inline -> r) -> SlamDown -> r
 everything b i = runIdentity <<< everythingM (pure <<< b) (pure <<< i)
 
-eval :: (Maybe String -> List String -> String) -> SlamDown -> SlamDown
-eval f = everywhere b i
+-- | Evaluates code blocks embedded within a SlamDown document.
+-- | - The first function handles the evaluation of fenced code blocks and
+-- |   inline code values. The first argument is the type associated with a
+-- |   fenced code block, taken from the opening triple-backtick. The second
+-- |   argument is each line of the code block. In the case of inline code the
+-- |   first argument is always `Nothing` and the `List` will only contain one
+-- |   item.
+-- | - The second function handles code blocks used to provide a default value
+-- |   for a text-based input.
+-- | - The third function handles singular values used in determining the
+-- |   selected value in a dropdown or collection of radio buttons.
+-- | - The fourth function handles lists of values used for the list of options
+-- |   in a dropdown, the list of values for checkboxes, and the list of
+-- |   selected values for checkboxes.
+eval :: forall m. (Monad m)
```

Comment:
> There may be some benefit in the functorified approach though, as then you'd know whether a piece of SlamDown has been evaluated or not, as it could flip from SlamDown Void to SlamDown (HTML p i) or something like that.

Makes sense but I'm happy to merge this now, I know you're trying to get this in by Monday. 

Merge now, revisit later??